liquidhaskell-boot/src/Language/Haskell/Liquid/Bare/Axiom.hs

```diff
@@ -348,6 +356,26 @@ makeAssumeType allowTC tce lmap dm sym mbT v def
     ref        = F.Reft (F.vv_, F.PAtom F.Eq (F.EVar F.vv_) le)
     mkErr s    = ErrHMeas (sourcePosSrcSpan $ loc sym) (pprint $ val sym) (PJ.text s)
     bbs        = filter isBoolBind xs
+
+    -- rTypeSortExp produces monomorphic sorts from polymorphic types.
```

Comment:
At this point, why not go a step further and fix `rTypeSortExp` so it doesn't produce `FObj` for type variables?

Reply:
I don't know if it breaks stuff somewhere else, for example when I made that modification I broke coercions/GADTs as inside PLE as the code that handled them was only taking care of the fake type variables and not of the real ones